src/utils/url-utils.ts

```diff
@@ -57,3 +57,35 @@ const prepareTilesetUrl = (parsedUrl) => {
     .concat("layers/0");
   return `${parsedUrl.origin}${replacedPathName}${parsedUrl.search}`;
 };
+
+export const viewStateToUrlParams = (viewState, setSearchParams) => {
+  const search = Object.fromEntries(
+    new URLSearchParams(window.location.search)
+  );
+  const { longitude, latitude, pitch, bearing, zoom } = viewState.main;
+  setSearchParams(
+    {
+      ...search,
+      longitude,
+      latitude,
+      pitch,
+      bearing,
+      zoom,
+    },
+    { replace: true }
+  );
+};
+
+export const urlParamsToViewState = (viewState, setStateUrlViewStateParams) => {
+  const search = new URLSearchParams(window.location.search);
+  const urlViewStateParams = {};
+  for (const viewStateParam of search) {
+    if (
+      Object.keys(viewState.main).includes(viewStateParam[0]) &&
+      !isNaN(parseFloat(viewStateParam[1]))
+    ) {
+      urlViewStateParams[viewStateParam[0]] = parseFloat(viewStateParam[1]);
+    }
+  }
+  setStateUrlViewStateParams(urlViewStateParams);
```

Comment:
Ditto: I prefer returning a value and setting the state in the component.

Reply:
Done